app/models/deploy_service.rb

```diff
@@ -122,4 +123,25 @@ def notify_outbound_webhooks(deploy)
   def send_sse_deploy_update(type, deploy)
     SseRailsEngine.send_event('deploys', type: type, deploy: DeploySerializer.new(deploy, root: nil))
   end
+
+  # TODO: make this work with remote images too ... and images where the local image no longer exists
```

Comment:
Here's a blog post explaining how you can add a tag to a docker image in a remote registry, without (necessarily) needing to do a `docker pull` and then a `docker push`: https://wheleph.gitlab.io/2016/07/13/tagging-using-docker-registry-http-api-v2/